awscli/s3transfer/crt.py

```diff
@@ -613,6 +605,96 @@ def invoke_all_callbacks(*args, **kwargs):
 
         return invoke_all_callbacks
 
+    def _get_make_request_args_put_object(
+        self,
+        request_type,
+        call_args,
+        coordinator,
+        future,
+        on_done_before_calls,
+        on_done_after_calls,
+    ):
+        send_filepath = None
+        if isinstance(call_args.fileobj, str):
+            send_filepath = call_args.fileobj
+            data_len = self._os_utils.get_file_size(send_filepath)
+            call_args.extra_args["ContentLength"] = data_len
+        else:
+            call_args.extra_args["Body"] = call_args.fileobj
+
+        # Suppress botocore's automatic MD5 calculation by setting an override
+        # value that will get deleted in the BotocoreCRTRequestSerializer.
+        # The CRT S3 client will automatically compute checksums as part of
+        # requests it makes.
```

Comment:
CRT S3 Client doesn't compute checksums unless you pass an [S3ChecksumConfig](https://github.com/awslabs/aws-crt-python/blob/fbf82f23cecb754beb6df7c1a30f11700774f2b9/awscrt/s3.py#L94). See me setting the checksum algorithm and location [here](https://github.com/boto/s3transfer/pull/277/files#diff-d7d493766dac8cf549533003bff769f7bc5d766645facdbe6c53d188284c953aR635-R640)

So this (and the currently released versions) aren't doing any checksums or MD5 for uploads

Reply:
Gotcha. Yeah one of the follow ups is to automatically set checksums for these requests as you had it in your original PR. For now, I'll just delete/update the second sentence in that comment.